adam/perception/perception_graph.py

```diff
@@ -501,7 +505,44 @@ def to_dot_node(pattern_node: "NodePredicate") -> str:
         dot_graph.render(output_file)
 
 
-@attrs(frozen=True, slots=True, eq=False)
+class DebugRenderingObject:
+    """
+        Helper callable class for debugging purposes. We render graph every hundred calls to match. The outputs can be
+        found in test/renders folder. This callable is called within GraphMatching.match
+    """
+
+    def __init__(self, graph_to_match_against: PerceptionGraph) -> None:
+        self.graph_to_match_against: PerceptionGraph = graph_to_match_against
+        self.calls_to_match_counter: int = 0
+        self.start_time = process_time()
+        self.time_window = 0
+        self.mod = 1
+
+    def __call__(
+        self, graph: DiGraph, graph_node_to_pattern_node: Dict[Any, Any]
+    ) -> None:
+        self.calls_to_match_counter += 1
+        current_time = process_time()
+        if (
+            self.calls_to_match_counter % self.mod == 0
+            and (current_time - self.start_time) > self.time_window
+        ):
+            perception_graph = PerceptionGraph(graph)
+            title = (
+                "graph_"
+                + str(id(self.graph_to_match_against))
+                + "_call_"
+                + str(self.calls_to_match_counter)
+            )
+            mapping = {k: "match" for k, v in graph_node_to_pattern_node.items()}
+            perception_graph.render_to_file(
+                graph_name=title,
+                output_file=Path("../renders/" + title),
```

Comment:
Never write to a path which depends on your programs current working directory. Make the user specify the directory to write to in the constructor.